pallets/automation-time/src/lib.rs

```diff
@@ -79,188 +81,16 @@ pub mod pallet {
-				self.execution_times.capacity() == other.execution_times.capacity() &&
-				self.execution_times.to_vec() == other.execution_times.to_vec()
-		}
-	}
-
-	impl<T: Config> Task<T> {
-		pub fn new(
-			owner_id: AccountOf<T>,
-			provided_id: Vec<u8>,
-			execution_times: BoundedVec<UnixTime, T::MaxExecutionTimes>,
-			action: Action<T>,
-		) -> Self {
-			let executions_left: u32 = execution_times.len().try_into().unwrap();
-			Self { owner_id, provided_id, execution_times, executions_left, action }
-		}
-
-		pub fn create_event_task(
-			owner_id: AccountOf<T>,
-			provided_id: Vec<u8>,
-			execution_times: BoundedVec<UnixTime, T::MaxExecutionTimes>,
-			message: Vec<u8>,
-		) -> Task<T> {
-			let action = Action::Notify { message };
-			Self::new(owner_id, provided_id, execution_times, action)
-		}
-
-		pub fn create_native_transfer_task(
-			owner_id: AccountOf<T>,
-			provided_id: Vec<u8>,
-			execution_times: BoundedVec<UnixTime, T::MaxExecutionTimes>,
-			recipient_id: AccountOf<T>,
-			amount: BalanceOf<T>,
-		) -> Task<T> {
-			let action = Action::NativeTransfer {
-				sender: owner_id.clone(),
-				recipient: recipient_id,
-				amount,
-			};
-			Self::new(owner_id, provided_id, execution_times, action)
-		}
-
-		pub fn create_xcmp_task(
-			owner_id: AccountOf<T>,
-			provided_id: Vec<u8>,
-			execution_times: BoundedVec<UnixTime, T::MaxExecutionTimes>,
-			para_id: ParaId,
-			currency_id: T::CurrencyId,
-			encoded_call: Vec<u8>,
-			encoded_call_weight: Weight,
-		) -> Task<T> {
-			let action = Action::XCMP { para_id, currency_id, encoded_call, encoded_call_weight };
-			Self::new(owner_id, provided_id, execution_times, action)
-		}
-
-		pub fn create_auto_compound_delegated_stake_task(
-			owner_id: AccountOf<T>,
-			provided_id: Vec<u8>,
-			execution_time: UnixTime,
-			frequency: Seconds,
-			collator_id: AccountOf<T>,
-			account_minimum: BalanceOf<T>,
-		) -> Task<T> {
-			let action = Action::AutoCompoundDelegatedStake {
-				delegator: owner_id.clone(),
-				collator: collator_id,
-				account_minimum,
-				frequency,
-			};
-			Self::new(owner_id, provided_id, vec![execution_time].try_into().unwrap(), action)
-		}
-
-		pub fn get_executions_left(&self) -> u32 {
-			self.executions_left
-		}
-	}
-
-	#[derive(Debug, Encode, Decode, TypeInfo)]
-	#[scale_info(skip_type_params(T))]
-	pub struct TaskHashInput<T: Config> {
-		owner_id: AccountOf<T>,
-		provided_id: Vec<u8>,
-	}
-
-	impl<T: Config> TaskHashInput<T> {
-		pub fn new(owner_id: AccountOf<T>, provided_id: Vec<u8>) -> Self {
-			Self { owner_id, provided_id }
-		}
-	}
+	pub type AccountTaskId<T> = (AccountOf<T>, TaskId<T>);
+	pub type ActionOf<T> = Action<AccountOf<T>, BalanceOf<T>, <T as Config>::CurrencyId>;
+	pub type TaskOf<T> = Task<
+		AccountOf<T>,
+		BalanceOf<T>,
+		<T as Config>::CurrencyId,
+		<T as Config>::MaxExecutionTimes,
+	>;
+	pub type MissedTaskV2Of<T> = MissedTaskV2<AccountOf<T>, TaskId<T>>;
```

Comment:
should these belong in the types file as well?

Reply:
No. They are the types w/ the generics from the pallet